src/DynamoCore/Models/HomeWorkspaceModel.cs

```diff
@@ -140,7 +147,84 @@ public override void Clear()
         {
             base.Clear();
             PreloadedTraceData = null;
-            RunEnabled = true;
+            RunSettings.Reset();
+        }
+
+        /// <summary>
+        /// Start periodic evaluation by the given amount of time. If there
+        /// is an on-going periodic evaluation, an exception will be thrown.
+        /// </summary>
+        /// <param name="milliseconds">The desired amount of time between two 
+        /// evaluations in milliseconds.</param>
+        /// <param name="context">A synchronization context belonging to the 
+        /// thread on which you want PulseMaker callbacks to execute.</param>
+        public void StartPeriodicEvaluation(int milliseconds, SynchronizationContext context)
+        {
+            if (pulseMaker == null)
+            {
+                pulseMaker = new PulseMaker(context);
+            }
+
+            pulseMaker.RunStarted += pulseMaker_RunStarted;
+            EvaluationCompleted += pulseMaker.OnRunExpressionCompleted;
+
+            if (pulseMaker.TimerPeriod != 0)
+            {
+                throw new InvalidOperationException(
+                    "Periodic evaluation cannot be started without stopping");
+            }
+
+            pulseMaker.Start(milliseconds);
+        }
+
+        private void pulseMaker_RunStarted(SynchronizationContext context)
+        {
+            var nodesToUpdate = Nodes.Where(n => n.EnablePeriodicUpdate);
+
+            context.Post(MarkNodesAsModifiedAndUpdate, nodesToUpdate);
```

Comment:
So, the only place where we obtain the `SynchronizationContext` is in the place where it's originally obtained?